### PR TITLE
Update Composer lock file and fix test failures

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Updates the composer lock file with i18n-mvc and flash messenger packages.

* Tests should still fail on lowest because of the namespace rename from `Laminas\Mvc\Router` to `Laminas\Router`
* The `UrlIntegrationTest` should still fail because the request retrieved from the container is not a `ConsoleRequest`

### CODE REVIEW

**Comment by gsteel** [comment]
The UrlIntegrationTest is problematic in that it is unclear to me why a legacy console request is expected. I believe that in the past we may have been able to guarantee a console request was present. Given that Console itself is deprecated, can this test be left in a flaky state, given that the expected output is still the same? I would imagine that the test will be deleted in the not too distant future anyhow… Paging @Ocramius for review :)

**Comment by Ocramius** [comment]
> it is unclear to me why a legacy console request is expected

Probably for testing purposes - the test is most likely there to ensure BC compatibility.

 > Given that Console itself is deprecated, can this test be left in a flaky state, given that the expected output is still the same?

Sadly not: deprecation is not removal of support.

 >  I would imagine that the test will be deleted in the not too distant future anyhow…

On the `3.0.0` release, we can certainly remove it, but in `2.x`, it must stay in place if it was ever green in the past.

File: test/Helper/Navigation/AbstractTest.php [hunk 1]
```diff
@@ -89,10 +86,6 @@ protected function setUp(): void
     {
         $cwd = __DIR__;
 
-        $this->routeMatchType = class_exists(V2RouteMatch::class)
```
Comment (Ocramius) [inline test/Helper/Navigation/AbstractTest.php:92]: This conditional indicates that we can deal with V2 **or** V3. Can we somehow reduce that via `composer.json` restrictions (support only one version)?

File: test/Helper/UrlIntegrationTest.php [hunk 2]
```diff
@@ -29,16 +27,15 @@
  */
 class UrlIntegrationTest extends TestCase
 {
+    private $serviceManager;
+
     protected function setUp(): void
     {
-        $this->literalRouteType = class_exists(V2HttpRoute\Literal::class)
```
Comment (Ocramius) [inline test/Helper/UrlIntegrationTest.php:34]: Similar to above: putting a restriction in `composer.json` is necessary for this conditional to be removed.

File: test/Helper/UrlTest.php [hunk 3]
```diff
@@ -52,39 +46,15 @@ class UrlTest extends TestCase
      */
     protected function setUp(): void
     {
-        $this->routeMatchType = class_exists(RouteMatch::class)
```
Comment (Ocramius) [inline test/Helper/UrlTest.php:55]: Similar discussion here: `Laminas\Router` dependency must be bumped

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: test/Helper/Navigation/AbstractTest.php (hunk 1 above)
Comment (gsteel) [reply test/Helper/Navigation/AbstractTest.php:92]: I've made the minimum version of Laminas\Navigation ^2.8.1 in dev which is the version where support was added for Laminas\Router vs Laminas\Mvc\Router.

Comment (Ocramius) [reply test/Helper/Navigation/AbstractTest.php:92]: Does `"laminas/laminas-router": "^3.0.1",` cover that then?

Comment (gsteel) [reply test/Helper/Navigation/AbstractTest.php:92]: I assumed that if --dev will only install v3, then it's simply not possible to test against v2 without changing the dev requirement of router to 2||3 or something like that - so 2.8.1 of navigation fixes the test failures and renders the checks for v2 router obsolete???

Comment (gsteel) [reply test/Helper/Navigation/AbstractTest.php:92]: P.S. I think that the dev requirement of navigation could actually be bumped 2.9.x due to 7.3 support getting added around that release

Comment (Ocramius) [reply test/Helper/Navigation/AbstractTest.php:92]: We can add a `"conflict": {"laminas/laminas-router": "<3.0.1"}` to solve that

**Comment by Ocramius** [comment]
This was manually merged, thanks @gsteel!

**Pull request closed without merging**